src/main/java/com/rultor/agents/github/qtn/QnReferredTo.java

```diff
@@ -82,20 +84,29 @@ public QnReferredTo(final String self, final Question qtn) {
     @Override
     public Req understand(final Comment.Smart comment,
         final URI home) throws IOException {
-        final String prefix = String.format("@%s ", this.login);
+        final String prefix = String.format("@%s", this.login);
         final Req req;
-        if (comment.body().trim().startsWith(prefix)) {
-            req = this.origin.understand(comment, home);
-        } else if (comment.body().contains(prefix)) {
-            new Answer(comment).post(
-                true,
-                String.format(
-                    QnReferredTo.PHRASES.getString("QnReferredTo.mentioned"),
-                    prefix
-                )
-            );
-            Logger.info(this, "mention found in #%d", comment.issue().number());
-            req = Req.DONE;
+        final Matcher matcher = Pattern.compile(
+            String.format("(?:^|(?:.*?(?:\\s|,)))(%s)\\b.*?", prefix)
```

Comment:
@original-brownbear right now this pattern is even more complex - can you extract it to constant and add a Javadoc with explanation to it?